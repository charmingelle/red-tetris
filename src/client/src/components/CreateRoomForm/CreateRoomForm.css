.create-room-form {
  width: 315px;
  display: flex;
  flex-direction: column;
}

.create-room-label {
  margin-bottom: 10px;
  font-size: 36px;
  color: #f9f9f9;
}

.enter-room-name-error {
  margin-bottom: 10px;
  font-size: 30px;
  color: #c00;
}

.create-room-input {
  height: 45px;
  margin-bottom: 10px;
  border: 5px solid;
  border-left-color: #b804e6;
  border-top-color: #b804e6;
  border-right-color: #660099;
  border-bottom-color: #660099;
  font-size: 36px;
}

.create-room-input:focus {
  outline: none;
}

.create-room-button {
  background-color: #9901cc;
  border: 5px solid;
  border-left-color: #b804e6;
  border-top-color: #b804e6;
  border-right-color: #660099;
  border-bottom-color: #660099;
  color: #fff;
  font-size: 30px;
  font-weight: bold;
  padding: 10px;
}

.create-room-button:focus {
  outline: none;
}
