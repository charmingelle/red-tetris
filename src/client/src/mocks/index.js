export const RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

export const LINE = [
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
];

export const GI = [
  [0, 0, 0],
  [1, 1, 1],
  [0, 0, 1],
];

export const EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];

export const NOT_EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const GAME_OVER_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, RED, RED, RED, RED, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const PILE_WITH_ROWS_TO_REMOVE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const TETRO = {
  figure: [GI],
  row: 5,
  col: 4,
  color: RED,
};

export const ROOM = {
  id: 'Room',
  leader: 'id-0',
  players: {
    'id-0': {
      id: 'id-0',
      name: 'Anna',
      pile: EMPTY_PILE,
      tetro: null,
      score: 0,
      isGameOver: false,
    },
  },
};

export const EMPTY_STORE = {
  socket: null,
  myId: null,
  myName: null,
  myRoom: null,
  people: {},
  rooms: {},
  tetro: null,
};

export const PERSON_ANNA = { name: 'Anna', score: 0 };

export const PERSON_OLYA = { name: 'Olya', score: 0 };
