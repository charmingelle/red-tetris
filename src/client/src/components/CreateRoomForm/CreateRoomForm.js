import React, { useState } from 'react';
import { connect } from 'react-redux';
import './CreateRoomForm.css';

const createRoom = (newRoomName, setNewRoomName, socket) => () => {
  if (newRoomName) {
    socket.emit('create-room', { roomId: newRoomName });
    setNewRoomName('');
  }
};

const changeNewRoomName = setNewRoomName => ({ target: { value } }) =>
  setNewRoomName(value);

const CreateRoomFormInner = ({ socket }) => {
  const [newRoomName, setNewRoomName] = useState('');

  return (
    <div className="create-room-form">
      <div className="create-room-label">Create own room</div>
      <input
        className="create-room-input"
        value={newRoomName}
        type="text"
        placeholder="Enter name..."
        onChange={changeNewRoomName(setNewRoomName)}
      />
      <button
        className="create-room-button"
        onClick={createRoom(newRoomName, setNewRoomName, socket)}
      >
        Create
      </button>
    </div>
  );
};

const mapStateToProps = ({ socket }) => ({ socket });

export const CreateRoomForm = connect(mapStateToProps)(CreateRoomFormInner);
