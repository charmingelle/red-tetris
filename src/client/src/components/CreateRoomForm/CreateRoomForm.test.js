import React from 'react';
import { CreateRoomForm } from './CreateRoomForm';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { EMPTY_STORE } from '../../mocks';

const mockStore = configureStore([]);

test('CreateRoomForm', () => {
  const store = mockStore(EMPTY_STORE);

  expect(
    renderer
      .create(
        <Provider store={store}>
          <CreateRoomForm />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
