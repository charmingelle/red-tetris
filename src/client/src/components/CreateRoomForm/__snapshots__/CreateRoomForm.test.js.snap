// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRoomForm 1`] = `
<div
  className="create-room-form"
>
  <div
    className="create-room-label"
  >
    Create own room
  </div>
  <input
    className="create-room-input"
    maxLength="15"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter name..."
    type="text"
    value=""
  />
  <button
    className="create-room-button"
    onClick={[Function]}
  >
    Create
  </button>
</div>
`;
